xsd/pointwise_models/zscore.py

```diff
@@ -0,0 +1,142 @@
+from sklearn.base import RegressorMixin
+from sklearn.linear_model.base import LinearModel
+import pandas as pd
+import numpy as np
+
+class ZScoreRegressor(LinearModel, RegressorMixin):
+    """ Z Score Regressor bias correction model wrapper
+
+    Apply a scikit-learn model (e.g. Pipeline) point-by-point. The pipeline
+    must implement the fit and predict methods.
+    """
+
+    def __init__(self, n_analogs=200, kdtree_kwargs={}, query_kwargs={}, lr_kwargs={}):
+
+        self.n_analogs = n_analogs
+        self.kdtree_kwargs = kdtree_kwargs
+        self.query_kwargs = query_kwargs
+        self.lr_kwargs = lr_kwargs
+
+        
+        
+    
+    def fit(self, X, y, window_width):
+        """ Fit Z-Score Model
+             
+
+        Parameters
+        ----------
+        X : pd.Series or pd.DataFrame, shape (n_samples, 1)
+            Training historical model data
+        y : pd.Series or pd.DataFrame, shape (n_samples, 1)
+            Target measured values.
+        window_width :  The size of the moving window for 
+            statistical analysis. Default is 31 days.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        
+        X_mean, X_std, X_zscore = _get_circ_stats(X, window_width)
+        y_mean, y_std, y_zscore = _get_circ_stats(y, window_width)
+        
+        shift, scale = _calc_correction_params(X_mean, X_std, y_mean, y_std)
+
+        self.shift = shift
+        self.scale = scale
+        
+        return self    
+        
+        
+    def predict(self, X, window_width=31, var_str='U'):
+        """ Predict
+        
+        This is where I change my future (X input is future raw data) based on fit
+        
+        Parameters
+        ----------
+        X : DataFrame, shape (n_samples, 1)
+            Samples.
+        window_width : The size of the moving window for 
+            statistical analysis. Default is 31 days.
+        var_str :  The key associated with the target 
+            dataframe variable
+
+        Returns
+        -------
+        corrected : pd.DataFrame, shape (n_samples, 1)
+            Returns corrected values.
+        """
+        
+        corrected = _z_score_correction(X, var_str, self.shift, self.scale, window_width)
+
+        
+        return corrected
+  
+
+
+#Helpers
+# Helpers for Fit
+def _hack_circular_rolling(df):
```

Comment:
let's add doc strings and comments to these helper functions

Reply:
On it